android/java/org/chromium/chrome/browser/playlist/hls_content/HlsServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+    private PlaylistService mPlaylistService;
+
+    class LocalBinder extends Binder {
+        HlsServiceImpl getService() {
+            return HlsServiceImpl.this;
+        }
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return mBinder;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_PLAYLIST)) {
+            initPlaylistService();
+        }
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        startHlsContentFromQueue();
+        return Service.START_NOT_STICKY;
+    }
+
+    private void startHlsContentFromQueue() {
+        PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK, () -> {
+            PlaylistRepository playlistRepository = new PlaylistRepository(mContext);
+            if (playlistRepository != null) {
+                HlsContentQueueModel hlsContentQueueModel =
+                        playlistRepository.getFirstHlsContentQueueModel();
+                if (hlsContentQueueModel != null && mPlaylistService != null) {
+                    String playlistItemId = hlsContentQueueModel.getPlaylistItemId();
+                    mPlaylistService.getPlaylistItem(playlistItemId, playlistItem -> {
+                        HlsUtils.getManifestFile(mContext, mPlaylistService, playlistItem,
+                                new HlsUtils.HlsManifestDelegate() {
+                                    @Override
+                                    public void onHlsManifestCompleted(
+                                            Queue<Segment> segmentsQueue) {
+                                        int total = segmentsQueue.size();
+                                        String hlsMediaFilePath =
+                                                HlsUtils.getHlsMediaFilePath(playlistItem);
+                                        HlsUtils.deleteFileIfExist(hlsMediaFilePath);
+                                        HlsUtils.getHLSFile(mContext, mPlaylistService,
+                                                playlistItem, segmentsQueue,
+                                                new HlsUtils.HlsFileDelegate() {
+                                                    @Override
+                                                    public void onProgress(int sofar) {
+                                                        PlaylistUtils.updateHlsContentProgress(
+                                                                new HlsContentProgressModel(
+                                                                        playlistItem.id,
+                                                                        (long) total, (long) sofar,
+                                                                        String.valueOf((sofar * 100)
+                                                                                / total)));
```

Comment:
perhaps a check for `total == 0` is needed to avoid dividing on 0.

Reply:
Good suggestion. i am adding a check